examples/seismic/skew_self_adjoint/operators.py

```diff
@@ -0,0 +1,330 @@
+# TODO figure out if can get time sampling from the SparseTimeFuntions src/rec
+# TODO replace the conditional logic in the stencil with comprehension
+from devito import Eq, Operator, Function, TimeFunction
+
+
+def iso_stencil(field, model, **kwargs):
+    """
+    Stencil for the scalar isotropic visco- acoustic variable density
+    skew self adjoint wave equation:
+
+        b/v^2 [ P.dt2 + w/Q P.dt ] = (b P.dx).dx + (b P.dy).dy + (b P.dz).dz + s
+
+    Note derivative shifts are omitted for simplicity above.
+    See implementation notebook ssa_01_iso_implementation1.ipynb for more details.
+
+    Parameters
+    ----------
+    field : TimeFunction, required
+        The pressure wavefield computed solution.
+    model : Dictionary <string>:<Function>, contains:
+        'b': Buoyancy = reciprocal density (units: m^3/kg)
+        'v': Velocity (units: m/msec or km/sec)
+        'wOverQ': The w/Q field for dissipation only attenuation.
+    forward : bool, optional
+        The propagation direction. Defaults to True.
+    q : TimeFunction, Function or float, optional
+        Full-space/time source of the wave-equation.
+
+    Returns
+    ----------
+    The time update stencil
+    """
+    # Get the Functions for buoyancy, velocity, and wOverQ
+    assert 'b' in model, "model dictionary must contain key 'b'"
+    assert 'v' in model, "model dictionary must contain key 'b'"
+    assert 'wOverQ' in model, "model dictionary must contain key 'wOverQ'"
+    b = model['b']
+    v = model['v']
+    wOverQ = model['wOverQ']
+
+    # Define time step of pressure wavefield to be updated
+    forward = kwargs.get('forward', True)
+
+    if forward:
+        field_next = field.forward
+        field_prev = field.backward
+    else:
+        field_next = field.backward
+        field_prev = field.forward
+
+    # Get the source
+    q = kwargs.get('q', 0)
+
+    # Define the time update equation for 2d/3d
+    # TODO: consider replacing conditional logic with @mloubout's comprehension
+    # space_fd = sum([getattr(b * getattr(field, 'd%s'%d.name)(x0=d+d.spacing/2)),
```

Comment:
FWIW, this "comprehension" would indeed make the code much smaller, but perhaps harder to penetrate ?

@mloubout knows that already -- I can't stress enough how `getattr` makes everything so awful, it should really be avoided at all costs. It's typically symptomatic of a design flaw in the code(base)

Reply:
so, do I get it correctly -- are we missing a laplacian in the DSL that allows evaluation at whatever `x0` ?

Reply:
No we are not missing a laplacian at x0, this is not what this does, and you can't get out of getattr (that's how .laplace/div/.... aare built too) as you need `f.d%s % dim.name` and this is not a valid syntaz withut the getattr.

Reply:
Other way is a lot of if/else with number of dimension whcih is not very nice

Reply:
> and you can't get out of getattr (that's how .laplace/div/.... aare built too) 

indeed. But here we're in user code, not in the compiler, so if there's no neat way of expressing "that", then it's a symptom we're missing something in the DSL

Reply:
remove the comment then?

Reply:
what math is this thing actually implementing? I'm only noticing now that the first derivative is at `x+x.spacing/2` while the latter at `x-x.spacing/2`

Reply:
see comment on slack